<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>KeyGenerator</class>
 <widget class="QDialog" name="KeyGenerator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>454</width>
    <height>205</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Key Generator</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="3" column="3">
    <widget class="QPushButton" name="IncludeSymbols">
     <property name="text">
      <string>! @ #</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">buttonGroup</string>
     </attribute>
    </widget>
   </item>
   <item row="2" column="0" colspan="4">
    <widget class="QTextEdit" name="Code"/>
   </item>
   <item row="3" column="0">
    <widget class="QPushButton" name="IncludeCapitals">
     <property name="text">
      <string>A-Z</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">buttonGroup</string>
     </attribute>
    </widget>
   </item>
   <item row="4" column="0" colspan="4">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="4">
    <widget class="QSlider" name="keyLengthSlider">
     <property name="minimum">
      <number>4</number>
     </property>
     <property name="maximum">
      <number>128</number>
     </property>
     <property name="sliderPosition">
      <number>8</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QPushButton" name="IncludeSmall">
     <property name="text">
      <string>a-z</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">buttonGroup</string>
     </attribute>
    </widget>
   </item>
   <item row="0" column="0" colspan="4">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="SafeLevel">
       <property name="font">
        <font>
         <family>Noto Sans</family>
         <pointsize>10</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QSpinBox" name="KeyLength">
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <property name="minimum">
        <number>4</number>
       </property>
       <property name="maximum">
        <number>128</number>
       </property>
       <property name="value">
        <number>8</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="Update">
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../images.qrc">
         <normaloff>:/images/update-document.svg</normaloff>:/images/update-document.svg</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="Copy">
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../images.qrc">
         <normaloff>:/images/edit-copy.svg</normaloff>:/images/edit-copy.svg</iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="3" column="2">
    <widget class="QPushButton" name="IncludeNumbers">
     <property name="text">
      <string>0-9</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">buttonGroup</string>
     </attribute>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../images.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>KeyGenerator</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>KeyGenerator</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="buttonGroup">
   <property name="exclusive">
    <bool>false</bool>
   </property>
  </buttongroup>
 </buttongroups>
</ui>
